bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/action/TelegramActions.java

```diff
@@ -394,6 +396,153 @@ public boolean sendTelegramPhoto(@ActionInput(name = "photoURL") @Nullable Strin
         return sendTelegramPhoto(photoURL, caption, null, null);
     }
 
+    @RuleAction(label = "send animation", description = "Send an Animation using the Telegram API.")
+    public boolean sendTelegramAnimation(@ActionInput(name = "animationURL") @Nullable String animationURL,
+            @ActionInput(name = "caption") @Nullable String caption) {
+        TelegramHandler localHandler = handler;
+        if (localHandler != null) {
+            for (Long chatId : localHandler.getReceiverChatIds()) {
+                if (!sendTelegramAnimation(chatId, animationURL, caption)) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    @RuleAction(label = "send animation", description = "Send an Animation using the Telegram API.")
+    public boolean sendTelegramAnimation(@ActionInput(name = "chatId") @Nullable Long chatId,
+            @ActionInput(name = "animationURL") @Nullable String animationURL,
+            @ActionInput(name = "caption") @Nullable String caption) {
+        if (animationURL == null) {
+            logger.warn("Animation URL not defined; unable to retrieve video for sending.");
+            return false;
+        }
+        if (chatId == null) {
+            logger.warn("chatId not defined; action skipped.");
+            return false;
+        }
+        TelegramHandler localHandler = handler;
+        if (localHandler != null) {
+            final SendAnimation sendAnimation;
+            if (animationURL.toLowerCase().startsWith("http")) {
+                // load image from url
```

Comment:
Do you need to download the url? I thought the telegram api supported sending urls as is.

Reply:
Yes for multiple reasons.

1. This is the way it was already done for sendPhoto
2. You get larger file sizes allowed if done this way. 20MB = 14 second long recording on my cameras.
3. The way I feel most people will use this is for sending a PRIVATE recording made from an IP camera that only exists inside a private network. This allows the video/photos and more to be sent out to your mobile phone without needing to open ports in a firewall. If the source file was externally accessible, then I would simply use a normal message and use a link in the text based message.
4. It will work in all use cases and keep code simple.

Quote from API docs....
Sending files
There are three ways to send files (photos, stickers, audio, media, etc.):

1. If the file is already stored somewhere on the Telegram servers, you don't need to reupload it: each file object has a file_id field, simply pass this file_id as a parameter instead of uploading. There are no limits for files sent this way.

2. Provide Telegram with an HTTP URL for the file to be sent. Telegram will download and send the file. 5 MB max size for photos and 20 MB max for other types of content.

3. Post the file using multipart/form-data in the usual way that files are uploaded via the browser. 10 MB max size for photos, 50 MB for other files.

Reply:
sounds good to me.